src/client/Client.cc

```diff
@@ -12824,13 +12849,20 @@ int Client::_mkdir(Inode *dir, const char *name, mode_t mode, const UserPerm& pe
   req->dentry_unless = CEPH_CAP_FILE_EXCL;
 
   mode |= S_IFDIR;
-  bufferlist xattrs_bl;
-  int res = _posix_acl_create(dir, &mode, xattrs_bl, perm);
+  bufferlist bl;
+  int res = _posix_acl_create(dir, &mode, bl, perm);
   if (res < 0)
     goto fail;
   req->head.args.mkdir.mode = mode;
-  if (xattrs_bl.length() > 0)
-    req->set_data(xattrs_bl);
+  if (is_snap_op) {
+    SnapPayload payload;
+    bl.clear();
```

Comment:
I'm looking into this failure:

````
2020-12-13T06:56:40.129 INFO:tasks.ceph.mds.b.smithi081.stderr:2020-12-13T06:56:40.129+0000 7fda1712e700 -1 failed to decode message of type 24 v4: End of buffer
````

From: /ceph/teuthology-archive/pdonnell-2020-12-13_06:09:39-fs-wip-pdonnell-testing-20201213.040840-distro-basic-smithi/5704914/teuthology.log

I wonder if this `bl.clear();` is the cause. It looks suspicious to me.

Reply:
I'll take a look

Reply:
you are right -- message type is CEPH_MGS_CLIENT_REQUEST (24, w/ header version 4). I'll fix the bug.

Reply:
@batrick mentioned that this is probably related to a kclient change.

Waiting for fos further updates.